### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by nonword**
Proposal for a generalized client for interacting with the Sierra REST API in Ruby

Supports authenticated GET, POST on arbitrary Sierra REST endpoints.

### CODE REVIEW

File: lib/sierra_api_client.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      raise SierraApiClientError.new(e), "Failed to GET #{path}: #{e.message}"
+    end
+
+    logger.debug "SierraApiClient: Got Sierra api response", { code: response.code, body: response.body }
+
+    parse_response response
+  end
+
+
+  def post (path, body, options = {})
```
Comment (danamansana) [inline lib/sierra_api_client.rb:54]: Minor inessential comment:

Are the differences in the way the post request and the get request are set up essential? It seems like the only thing that should really be different are 1.  setting up body/content type 2. the actual method.

Comment (siberry) [reply lib/sierra_api_client.rb:54]: As an end user, I do like having the separate `get` and `post` methods. I guess under the hood these methods could use the same method/share more helper methods? But seems  fine to me. Easy to read.

File: README.md [hunk 2]
```diff
@@ -1 +1,85 @@
...
+})
+```
+
+### Requests
+
+Example GET:
+
+```ruby
+bib_response = sierra_client.get 'patrons/1234'
+bib = bib_response.body
```
Comment (siberry) [inline README.md:36]: I have not worked with the Sierra API yet. Does the patrons endpoint return data related to bibs?

File: lib/sierra_api_client.rb (hunk 1 above)
Comment (nonword) [reply lib/sierra_api_client.rb:31]: Good notes. I've amended the PR to DRY the GET and POST implementation in a new, generic `do_request` private method, which does most of the work. The prior differences between their implementation were not meaningful.

File: README.md (hunk 2 above)
Comment (nonword) [reply README.md:36]: That's a typo. Good catch. It's now ` bib_response = sierra_client.get 'bibs/1234' `

File: lib/sierra_api_client.rb [hunk 3]
```diff
@@ -25,68 +25,61 @@ def initialize(config = {})
...
+    # Create HTTP::Get or HTTP::Post
+    request =  Net::HTTP.const_get(method.capitalize).new(uri.path, request_headers)
 
-      logger.debug "SierraApiClient: Posting to Sierra api", { uri: uri, body: body }
+    # Add bearer token header
+    request['Authorization'] = "Bearer #{@access_token}" if options[:authenticated]
 
-      # Add bearer token header
-      request['Authorization'] = "Bearer #{@access_token}" if options[:authenticated]
+    # Allow caller to modify the request before we send it off:
```
Comment (danamansana) [inline lib/sierra_api_client.rb:66]: This is cool, I've never seen this syntax before

**Pull request merged**